src/test/RemoteSimulator.test.ts

```diff
@@ -0,0 +1,73 @@
+import {
+    CONNECTION_SUCCESS_MSG,
+    CONNECTION_FAIL_MSG,
+} from "../simularium/RemoteSimulator";
+import FrontEndError from "../simularium/FrontEndError";
+import { TEST_CONNECTION_SETTINGS } from "../constants";
+import { RemoteSimulator } from "..";
+
+describe("RemoteSimulator", () => {
+    // Silence console.debug messages like this in Jest output:
+    // "[netconnection] WS Connection Request Sent:  wss://..."
+    jest.spyOn(global.console, "debug").mockImplementation(() => jest.fn());
+
+    describe("connectToUri", () => {
+        test("creates a valid WebSocket object", () => {
+            const simulator = new RemoteSimulator({});
+            expect(simulator.socketIsValid()).toBe(false);
+
+            simulator.connectToUri(simulator.getIp());
```

Comment:
is this testing a real live internet connection?

Reply:
Yeah, it's actually connecting to our staging server. Is that ok?

Reply:
It just means you're no longer testing just simularium-viewer, you're now testing that the staging server and the internet connectivity to it are both alive and well.  There are so many reasons why it could fail.  I would resist doing this in the unit tests here.  Better if you can mock it somehow.  But if others prefer to have this be a true integration test, we can think about it that way.

Reply:
I think it's already mocked in the DummyRemoteSimulator ?

Reply:
Yeah that's a good point. I'll look into using the DummyRemoteSimulator or using Jest to mock some return values like I did in what I called the "key test number 2". I think I would prefer the second option because wouldn't it be better to test the RemoteSimulator class (even if some of its methods are mocked) rather than the DummyRemoteSimulator?

Reply:
I tend to agree, what you really want to mock is the back end, and not the RemoteSimulator itself.  Which i suppose is what the DummyRemoteSimulator is supposed to be trying to do, but it's extending RemoteSimulator.